File: frontend/src/api/item.ts
import client from './client'
import type { ItemSummary } from './types'

// 商品 API：提供商品摘要的单条与批量查询
export const itemApi = {
  summary: (itemId: string) =>
    client.get<ItemSummary>(`/api/items/${itemId}/summary`).then((r) => r.data),

  batch: (ids: string[]) =>
    client.get<Record<string, ItemSummary>>('/api/items/batch', { params: { ids: ids.join(',') } }).then((r) => r.data),

  // 刷新单个商品详情：采集详情页并更新销售状态
  // 触发场景：链接点击异步刷新、官方采集、抢单失败回退
  // taskId 可选：items 表无记录时回填 task_links.display 使用
  // 返回完整采集字段：实时模式下前端用此结果直接更新 liveItemsRef，避免重新搜索覆盖采集结果
  refresh: (itemId: string, taskId?: string) =>
    client.post<{
      ok: boolean
      item_id: string
      is_sold: boolean
      title: string
      price: number
      brand: string
      seller_id: string
      region: string
      want_cnt: number
      view_cnt: number
      thumb_url: string
      image_urls: string[]
      publish_time: string | null
      seller_nick: string
      seller_credit: number | null
    }>(
      `/api/items/${itemId}/refresh`,
      null,
      { params: taskId ? { task_id: taskId } : {} },
    ).then((r) => r.data),
}

File: backend/xianyu_hunter/domain/item.py
"""领域模型：商品"""
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class ItemSummary:
    """搜索结果列表中的商品摘要"""
    id: str
    title: str
    price: float
    region: str = ""
    brand: str = ""  # 商品品牌（从搜索 API 或标题兜底推断）
    seller_id: str = ""
    seller_nick: str = ""  # 卖家昵称（从搜索结果提取）
    want_cnt: int = 0
    view_cnt: int = 0
    publish_time: datetime | None = None
    thumb_url: str = ""
    is_sold: bool = False  # 是否已售出（用于前端过滤）
    # 从搜索API提取的卖家基本信息（用于降级评估）
    seller_credit_score: int | None = None  # 芝麻信用分
    seller_on_sale_count: int = 0  # 在售商品数
    seller_sold_count: int = 0  # 已售商品数
    seller_credit: str = ""  # 卖家信用度描述（如"极好/良好/优秀"，从搜索卡片提取）


@dataclass
class ItemDetail(ItemSummary):
    """商品详情"""
    description: str = ""
    image_urls: list[str] = field(default_factory=list)
    raw_json: dict = field(default_factory=dict)
    # 从详情页DOM提取的额外卖家信息（用于降级评估）
    detail_seller_nick: str = ""       # 详情页中的卖家昵称
    detail_credit_score: int | None = None  # 详情页中的信用分
    detail_on_sale_count: int = 0      # 在售数（从详情页）
    detail_sold_count: int = 0         # 已售数（从详情页）
    detail_register_days: int = 0      # 注册天数（从详情页"来闲鱼X天"）
